pkg/executor/util/utils.go

```diff
@@ -146,3 +145,27 @@ func NewNoopExecutors(config noop_executor.ExecutorConfig) executor.ExecutorProv
 	noopExecutor := noop_executor.NewNoopExecutorWithConfig(config)
 	return model.NewNoopProvider[model.Engine, executor.Executor](noopExecutor)
 }
+
+type PluginExecutorOptions struct {
+	Plugins []PluginExecutorConfig
+}
+
+func NewPluginExecutorProvider(
+	ctx context.Context,
+	cm *system.CleanupManager,
+	pluginOptions PluginExecutorOptions,
+) (executor.ExecutorProvider, error) {
+	pe := NewPluggableExecutor()
+	for _, cfg := range pluginOptions.Plugins {
+		if err := pe.RegisterPlugin(cfg); err != nil {
+			return nil, err
+		}
+	}
+	if err := pe.Start(ctx); err != nil {
+		return nil, err
+	}
+
+	cm.RegisterCallbackWithContext(pe.Stop)
```

Comment:
I am hoping to minimize our usage of cleanup manager inside different places of the code. I am hoping that we limit its usage to `/pkg/node` package and have better control on the sequence on how we close our components

Reply:
I am following the same pattern use by the `New*Provider` methods in the package, is that alright for now or should I do something different?

I'd like to remove usage of the CleanupManager completely, and I think we can achieve that using something like FX https://uber-go.github.io/fx/lifecycle.html#lifecycle-hooks